anime_downloader/gui/gui.py

```diff
@@ -0,0 +1,94 @@
+import eel
+from anime_downloader.sites import get_anime_class
+from anime_downloader.config import Config
+from anime_downloader import util
+from anime_downloader.sites.anime import AnimeEpisode
+import subprocess
+import pathlib
+
+@eel.expose
+def populate_dropdown():
+    from anime_downloader.sites import ALL_ANIME_SITES
+    providers = [i[1] for i in ALL_ANIME_SITES]
+    eel.populate_dropdown(providers)
+
+
+@eel.expose
+def anime_search(query, provider):
+    print('serching')
+    print(query, provider)
+    anime_cls = get_anime_class(provider)
+    print(anime_cls)
+    res = anime_cls.search(query)
+    print(res)
+
+    eel.populate_search([dict(title=i.title, url=i.url, poster=i.poster) for i in res])
+
+@eel.expose
+def get_animeinfo(url):
+    global anime
+    print('getting info of: '+ url)
+    anime = get_anime_class(url)(url)
+    print(anime)
+    print(anime[0].source().stream_url)
+    return {
+        'title': anime.title,
+        'length': len(anime),
+    }
+
+def play_mirror_file(player, title, file, ):
+    if player == 'mpv':
+        p = subprocess.Popen([
+            player,
+            '--title={}'.format(title),
+            file,
+            ])
+    else:
+        p = subprocess.Popen([ player, file
+            ])
```

Comment:
This is weird formatting. Also either use the mpv class from the `players` or use the one from util. I prefer former.